src/main/java/org/apache/commons/validator/routines/checkdigit/CASNumberCheckDigit.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.validator.routines.checkdigit;
+
+import org.apache.commons.validator.routines.CodeValidator;
+
+/**
+ * Modulus 10 <b>CAS Registry Number</b> (or <b>Chemical Abstracts Service</b> (CAS RN)) Check Digit
+ * calculation/validation.
+ *
+ * <p>
+ * CAS Numbers are unique identification numbers used
+ * to identify chemical substance described in the open scientific literature.
+ * </p>
+ *
+ * Check digit calculation is based on <i>modulus 10</i> with digits being weighted
+ * based on their position (from right to left).
+ *
+ * The check digit is found by taking the last digit times 1, the preceding digit times 2,
+ * the preceding digit times 3 etc., adding all these up and computing the sum modulo 10.
+ * For example, the CAS number of water is 7732-18-5:
+ * the checksum 5 is calculated as (8×1 + 1×2 + 2×3 + 3×4 + 7×5 + 7×6) = 105; 105 mod 10 = 5.
+ *
+ * <p>
+ * For further information see
+ *  <a href="https://en.wikipedia.org/wiki/CAS_Registry_Number">Wikipedia - CAS Registry Number</a>.
+ * </p>
+ *
+ * @since 1.9
+ */
+public final class CASNumberCheckDigit extends ModulusCheckDigit {
+
+    private static final long serialVersionUID = -5387334603220786657L;
+
+    /** Singleton Check Digit instance */
+    public static final CheckDigit CAS_CHECK_DIGIT = new CASNumberCheckDigit();
```

Comment:
Use `INSTANCE` for the singleton name, make it private and add a `getInstance()` public static method (like `DoubleValidator.getInstance()` and friends.)